internal/charmstore/stats.go

```diff
@@ -587,6 +601,20 @@ func (s *Store) IncrementDownloadCounts(id *router.ResolvedURL) error {
 	if err := s.IncCounter(key); err != nil {
 		return errgo.Notef(err, "cannot increase stats counter for %v", key)
 	}
+	if id.PromulgatedRevision == -1 {
+		// Check that the id really is for an unpromulgated entity
+		entity, err := s.FindEntity(id, "promulgated-revision")
```

Comment:
It's a bit of a pity about this extra round trip, especially when the only place that calls IncrementDownloadCounts has already retrieved the entity in question. It's not immediately clear how to make only one request and keep the code nicely structured, but perhaps worth a comment.